### ISSUE TITLE
Black screen OpenGL

### ISSUE THREAD

**Issue by DuFF14** [OSVR/OSVR-Unity-Rendering#5]
Edited OP since it is no longer relevant. Current state of this issue is that OpenGL Unity-Rendering works on Windows in extended mode. There is a black screen in direct mode. Next step is to get it working on non-Windows platforms in extended mode.

**Comment by DuFF14** [comment]
Original post is extremely old and details there can be ignored, but this issue still remains. Currently, PresentRenderBuffers returns false. Haven't tracked down why it returns false. There might be an issue here: https://github.com/OSVR/OSVR-Unity-Rendering/blob/openGL_RenderManager/OsvrRenderingPlugin.cpp#L548

**Comment by DuFF14** [comment]
We'll need a "MakeDevice" path on OpenGL RenderManager. With D3D, we pass Unity's D3D graphics device and context to RenderManager. We need to share Unity's OpenGL context with RenderManager. This is on the backburner for now.

**Comment by DuFF14** [comment]
There are a few different branches trying to get this working, but [oglSupport](https://github.com/OSVR/OSVR-Unity-Rendering/tree/oglSupport) branch has my most recent attempts at getting this to work. The oglSupport branch is most similar to [OpenGLSharedContextPresentExample](https://github.com/sensics/OSVR-RenderManager/blob/master/examples/RenderManagerOpenGLSharedContextPresentExample.cpp). Current status is that I can get one frame rendered to the display in extended mode, but its not updating with tracking. After about a second, the Unity editor crashes with stack trace:
RtlLookupFunctionEntry returned NULL function. Aborting stack walk.0x000000005283BA99 (nvoglv64) vk_optimusGetInstanceProcAddr

In a build, it doesn't crash, but in the output log I get an error on PresentRenderBuffers:
OPENGL NATIVE PLUG-IN ERROR: GL_INVALID_OPERATION: Operation illegal in current state.

This is blocking the path for getting RenderManager support into Unity on Linux. Any help on this would be appreciated. Someone proficient in OpenGL should take a look as I am no expert. Not sure how close a solution is. 
@ChristophHaag

**Comment by ChristophHaag** [comment]
That's a really windows centric implementation in oglSupport though. Lots of wgl functions, DWORD, HGLRC, ...

Perhaps `SDL_GL_SetAttribute(SDL_GL_SHARE_WITH_CURRENT_CONTEXT, 1);` could be used?

**Comment by russell-taylor** [comment]
That flag is being set.

**Comment by DuFF14** [comment]
@ChristopherMitchell oglSupport branch is working now in the editor, but not builds, and only once before crashing the editor. Couldn't get it to work without wglSharedLists, however.

**Comment by DuFF14** [comment]
Black screen is now only limited to direct mode.
Extended mode works in editor and build, but only for one run.
Remaning issues:
- Platform specific context sharing. Currently, this will only work on Windows.
- proper shutdown/init so that we can run Unity multiple times in the editor without crashing. I think the issue is again with context sharing. Context sharing is happening on graphics device init (when Unity starts up), but when the game runs and quits and runs again, it looks like it has a new context which does not get shared with the Rendering Plugin again.
- direct mode

**Comment by DuFF14** [comment]
Black screen is no longer the issue, please see: https://github.com/OSVR/OSVR-Unity-Rendering/issues/35
